### DISCUSSION TITLE
FAQ

### DISCUSSION THREAD

**Discussion by MithunKinarullathil** [MOV-AI/movai-flow#205] [Announcements]
MOV.AI Flow™ FAQ: 

1. What is MOV.AI Flow?
   1. A new IDE that introduces visualization and structure to ROS.
2. Why is it “source available” and not an open source?
   1. The goal of this license is to ensure that both MOV.AI and the ROS community can continue to grow and thrive side by side. 
   2. The idea behind the license is the users can use the MOV.AI code freely and adapt it to their needs. It can be used to develop commercial products with the exclusion of an IDE. They cannot provide or sell a competing IDE that was developed with and is based on MOV.AI source code.
3. Who is this tool for?
   1. The tool is intended for any Robot developer, whether commercial, academic or hobbyist who wants to develop a robot system, component, or algorithm, either simulated or physical. 
   2. The tool can be useful for both advanced users and beginners, as the drag-and-drop system speeds up the development time, without sacrificing flexibility and low-level coding if needed. 
4. Do I have to be proficient with ROS in order to use the tool? 
   1. No, not at all. There are multiple demos included in the SW that can help the user understand how projects are built and learn from them. In fact, Flow allows you to incorporate (existing) complex ROS code into your applications by simple drag-and-drop blocks of code.
5. Why should I use the tool and not directly work with ROS and a standard IDE? 
   1. In fact, you can still do it, AND you can use Flow. To build C++-based ROS code/nodes you can continue using the same tools.
   2. When building a project, you should use the new IDE if you want to enjoy the features we are bringing, namely build State Machines or reuse existing projects and algorithms.
6. How do I install MOV.AI Flow?
    1. You can follow the [instructions on our documentation website](https://flow.mov.ai/docs/installation).
    2. Stay tuned for updates via the ROS/Gazebo discourse group.
7. Where can I learn how to use it?
    1. You can follow our [documentation website](https://flow.mov.ai/docs) for understanding how to use it.
8. Where can I ask questions or doubts?
    1. You can search or post any questions on our [GitHub discussions](https://github.com/MOV-AI/movai-flow/discussions) site.
9. There are already tools to visualize and Debug ROS, how Flow is different? 
   1. We believe our approach is the cleanest, most intuitive option out there to build and manage ROS-based projects. Ideally, you should use other community tools along Flow to improve your experience.
10. What version of ROS dos MOV.AI Flow™ currently support?
   1. We currently support ROS noetic.
11. Do you support ROS2? If not, when will it be available? 
   1. We don’t currently support ROS2, but we are planning for it.
12. How can I connect this tool to my existing ROS project? 
   1. You can do this in two ways as listed below.
   2. Use host networking mode to connect to your project using ROS topics and services:
       1. When launching MOV.AI Flow™ for the first time you have an option to choose two modes of networking. If you select the host network mode, the ROS topics are shared between your computer and MOV.AI. You can find further information here.
   3. Importing your project into MOV.AI Flow™:
       1. Since the core of a Flow project is the state machine, the roslaunch file cannot be directly translated into Flow.
       2. Therefore, if you want to start using all the existing features of Flow, you need to drag and drop ROS nodes into the GUI, and link them smartly to launch your ROS project.
       3. Therefore, each ROS node has to be imported to the tool before you start building your state machines.
13. How can I deploy the flow into my real robot?
    1. Install Flow in your robot. Instead of Sim Drivers, replace those by the actual ROS nodes that you currently use to  connect to your HW (eg. ethernet, USB, etc)
14. Can I use it to control robots of different types? 
    1. Yes, Flow is generic enough to allow you to build behaviors for many different robots in the same way ROS allows you to do it.
15. Is the tool dependent on any specific HW (for either development / deployment on a Robot)? 
    1. At the moment we are supporting only x64 architecture.
16. Can I import my code or external ROS packages into Flow? 
    1. Underneath Flow, the ROS code is running. Therefore you should be able to install and run all your ROS code/nodes using Flow.
17. Can I import my roslaunch?
    1. Currently we don’t support direct import of roslaunch files. However you can follow our documentation for importing nodes.
18. Is it possible to import non-ROS executables to Flow? 
    1. By design, you should be able to run any executable in a similar manner as we launch ROS binaries. However,  we’ve never tested it thoroughly in our projects.
19. Which simulated models and environments do you provide? Can I add my own? 
    1. You should be able to load existing Models and Worlds from Fuel repo or from other sources just fine since we are working with stock Gazebo Ignition Fortress.
    2. You can definitely edit or build new models from scratch ([Ignition Fortress documentation](https://gazebosim.org/docs/fortress/tutorials)).
20. Why not improve roslaunch, instead of replacing it ?
    1. Roslaunch is limited in what you can do. There’s not really the concept of States and Transitions, unless you hack it with scripts. 
    2. It is still possible to use roslaunch in parallel with Flow. However, the ideal situation is to create projects in Flow with a holistic approach.
21. Can we run roscore and some nodes on one PC and talk to Flow on another PC?
    1. Yes you can. If you have chosen the host networking mode then it is very close to running your roscore on the host machine. [Documentation](https://flow.mov.ai/docs/launching-movai-1).
    2. From here onwards you can follow the [ROS tutorial for inter-robot communication](http://wiki.ros.org/ROS/Tutorials/MultipleMachines).
22. How do I add my nodes or launch files?
    1. Our [documentation page](https://flow.mov.ai/docs/importing-a-ros-node) explains in detail how you can do such thing. In brief, first you need to have your code built and then you create a new template (Visual component) for the node in IDE.
23. Where do I start?
    1. You can start by following the [installation instructions](https://flow.mov.ai/docs/installation).
    2. Once you have the software installed, you can follow our [documentation](https://flow.mov.ai/docs).
24. Can I follow ROS tutorials in MOV.AI Flow™?
    1. Even though the functionality of the tutorials can be replicated. There steps from ROS tutorials need modifications to have the same functionality in MOV.AI Flow™.
25. What are the system requirements?
    1. You can find it in the [supported systems](https://flow.mov.ai/docs/installation#supported-systems) section of our documentation website.
26. Can I install MOV.AI Flow™ on real hardware?
    1. Yes you can, given the system requirements are met.
27. Can I install MOV.AI Flow™ on raspberry pi?
    1. We currently provide no support for installing MOV.AI Flow™.
28. Can I install MOV.AI Flow™ on windows?
    1. Even though installing in Windows is not supported, you are welcome to try to get support in the [GitHub discussions](https://github.com/MOV-AI/movai-flow/discussions).
29. How to integrate a new robot in MOV.AI Flow™?
    1. You can follow our [documentation page](https://flow.mov.ai/docs/design-your-robot).
30. Can I create a trajectory using the UI?
    1. The MOV.AI Flow™ is limited in this regard. However we do have a UI  that supports drawing trajectories in the enterprise edition. You can get in [touch with our team](https://www.mov.ai/contact-us/) for further details.
31. How does the link work?
    1. A link in the MOV.AI Flow™ Flow editor represents a remap in the ROS echo system.
    2. The final remap name is calculated based on:
        1. The nodes it connects to,
        2. The port names it connects to, and
        3. Other links which share a common port to this link.
32. In what order the nodes are launched the flow editor?
    1. If you have a transition link between two MOV.AI Flow™ state nodes, then when the transition happens:
        1. First, the nodes of the live node is killed.
        2. Then, the dependent nodes of the next state node is launched.
        3. Then, the next state node is launched.
    2. In the above case, the order of the dependent nodes launched are random.
33. Can we check the resource consumption of each node?
    1. We currently do not have a mechanism to check this. However we do have the functionality in our enterprise edition.
34. What’s the debug procedure looks like?
    1. All the nodes are launched in the spawner-* docker container.
    2. We check the logs using docker logs command. You can find more details of docker log commands [here](https://docs.docker.com/engine/reference/commandline/logs/).
35. What about simulation?
    1. We ship Ignition Fortress with our software and includes two robots and two worlds to be used in the demos.
36. How is MOV.AI Flow™ Flow editor different from launch files in ROS?
    1. ROS launch files are a list of nodes and other ROS launch files, which are launched in no particular order when we call the ros launch command. In the Flow editor we control the sequence of launch and death of nodes similar to a state machine.
    2. Once a flow is started, the system determines the order of launching/killing the nodes. This is determined by the type of the links connected between nodes and state nodes if any.
37. How to uninstall MOV.AI Flow™?
    1. You can uninstall MOV.AI Flow™ by following this [guide](https://flow.mov.ai/docs/installation#uninstall).

**Comment by marcusvinicius178** [comment]
Hi @MithunKinarullathil . How to properly launch MoV-AI to make application work 
I have fullfilled all the requirements (OS, docker, disk space and so on)

`$ movai-flow-launch 

![image](https://user-images.githubusercontent.com/43826438/232917787-645f6919-8e49-45e1-a48f-c5398adcc511.png)


But I can't see the interface of MOVAI

![image](https://user-images.githubusercontent.com/43826438/232914962-c34289cd-2a05-4d1d-879a-ef05c2837f91.png)

**Reply by MithunKinarullathil** [reply]
Hello, @marcusvinicius178. Do you mind posting this in the issues section? Normally, it takes some time to download the docker images based on your internet speed. Can you share the `docker ps` results with us?
